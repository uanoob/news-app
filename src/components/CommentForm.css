.input-error {
  border: 1px solid red;
}

input:focus {
  outline: none;
}
