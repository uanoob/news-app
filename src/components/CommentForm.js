import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Send from '@material-ui/icons/Send';
import { checkValidityLength } from '../utils/validator.utils';
import { createComment } from '../store/actions';

const styles = theme => ({
  container: {
    display: 'flex',
    flexDirection: 'column',
    flexWrap: 'wrap',
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
  },
  button: {
    margin: theme.spacing.unit,
  },
  rightIcon: {
    marginLeft: theme.spacing.unit,
  },
});

class CommentForm extends Component {
  state = {
    comment: '',
    commentInputTouched: false,
    commentInputValid: false,
  };

  handleTextInput = (e) => {
    this.setState({
      comment: e.target.value,
      commentInputTouched: true,
      commentInputValid: checkValidityLength('comment', e.target.value),
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const {
      articleId,
      onCreateComment,
      userId,
      userName,
      handleDialogClick,
    } = this.props;
    const { comment } = this.state;
    onCreateComment(articleId, comment, userId, userName);
    this.clearTextField();
    handleDialogClick();
  };

  handleCancelClick = () => {
    const { handleDialogClick } = this.props;
    this.clearTextField();
    handleDialogClick();
  };

  clearTextField = () => {
    this.setState({
      comment: '',
      commentInputTouched: false,
      commentInputValid: false,
    });
  };

  render() {
    const { classes, dialog, handleDialogClick } = this.props;
    const { comment, commentInputTouched, commentInputValid } = this.state;

    return (
      <Dialog
        open={dialog}
        onClose={handleDialogClick}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Add a comment</DialogTitle>
        <DialogContent>
          <DialogContentText>
            To leave a comment to this article, please enter your comment here.
          </DialogContentText>
          <TextField
            autoFocus
            id="outlined-multiline-flexible"
            label="Your comment here"
            multiline
            rows="4"
            value={comment}
            onChange={event => this.handleTextInput(event)}
            className={classes.textField}
            margin="normal"
            variant="outlined"
            error={!commentInputValid && commentInputTouched}
            fullWidth
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={this.handleCancelClick} color="primary">
            Cancel
          </Button>
          <Button
            variant="contained"
            color="primary"
            className={classes.button}
            type="submit"
            onClick={this.handleSubmit}
            disabled={!commentInputValid}
          >
            Send
            <Send className={classes.rightIcon}>send</Send>
          </Button>
        </DialogActions>
      </Dialog>
    );
  }
}

CommentForm.propTypes = {
  classes: PropTypes.shape({
    container: PropTypes.string.isRequired,
    textField: PropTypes.string.isRequired,
  }).isRequired,
  articleId: PropTypes.string.isRequired,
  onCreateComment: PropTypes.func.isRequired,
  userId: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  dialog: PropTypes.bool.isRequired,
  handleDialogClick: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  userId: state.user.user._id,
  userName: state.user.user.name,
});

const mapDispatchToProps = {
  onCreateComment: createComment,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withStyles(styles)(CommentForm));
