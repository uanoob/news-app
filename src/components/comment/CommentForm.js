import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Send from '@material-ui/icons/Send';
import { checkValidityLength } from '../../utils/validator.utils';
import { createComment, updateComment } from '../../store/actions';

const styles = theme => ({
  container: {
    display: 'flex',
    flexDirection: 'column',
    flexWrap: 'wrap',
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
  },
  button: {
    margin: theme.spacing.unit,
  },
  rightIcon: {
    marginLeft: theme.spacing.unit,
  },
});

export class CommentFormComponent extends Component {
  state = {
    comment: '',
    commentInputTouched: false,
    commentInputValid: false,
  };

  handleTextInput = (e) => {
    this.setState({
      comment: e.target.value,
      commentInputTouched: true,
      commentInputValid: checkValidityLength('comment', e.target.value),
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const {
      articleId,
      onCreateComment,
      onUpdateComment,
      userId,
      userName,
      handleDialogClick,
      commentId,
    } = this.props;
    const { comment } = this.state;
    if (!commentId) {
      onCreateComment(articleId, comment, userId, userName);
    } else {
      onUpdateComment(articleId, comment, commentId);
    }
    this.clearTextField();
    handleDialogClick();
  };

  handleCancelClick = () => {
    const { handleDialogClick } = this.props;
    this.clearTextField();
    handleDialogClick();
  };

  clearTextField = () => {
    this.setState({
      comment: '',
      commentInputTouched: false,
      commentInputValid: false,
    });
  };

  render() {
    const {
      classes,
      dialog,
      handleDialogClick,
      title,
      description,
      commentText,
    } = this.props;
    const { comment, commentInputTouched, commentInputValid } = this.state;

    return (
      <Dialog
        open={dialog}
        onClose={handleDialogClick}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">{title}</DialogTitle>
        <DialogContent>
          <DialogContentText>{description}</DialogContentText>
          <TextField
            autoFocus
            id="comment-textfield-input"
            label="Your comment here"
            multiline
            rows="4"
            value={comment || commentText}
            onChange={event => this.handleTextInput(event)}
            className={classes.textField}
            margin="normal"
            variant="outlined"
            error={!commentInputValid && commentInputTouched}
            fullWidth
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={this.handleCancelClick} color="primary">
            Cancel
          </Button>
          <Button
            id="comment-submit-button"
            variant="contained"
            color="primary"
            className={classes.button}
            type="submit"
            onClick={this.handleSubmit}
            disabled={!commentInputValid}
          >
            Send
            <Send className={classes.rightIcon}>send</Send>
          </Button>
        </DialogActions>
      </Dialog>
    );
  }
}

CommentFormComponent.defaultProps = {
  commentId: '',
  commentText: '',
};

CommentFormComponent.propTypes = {
  classes: PropTypes.shape({
    container: PropTypes.string.isRequired,
    textField: PropTypes.string.isRequired,
  }).isRequired,
  articleId: PropTypes.string.isRequired,
  onCreateComment: PropTypes.func.isRequired,
  onUpdateComment: PropTypes.func.isRequired,
  userId: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  dialog: PropTypes.bool.isRequired,
  handleDialogClick: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  commentId: PropTypes.string,
  commentText: PropTypes.string,
};

const mapStateToProps = state => ({
  userId: state.user.user._id,
  userName: state.user.user.name,
});

const mapDispatchToProps = {
  onCreateComment: createComment,
  onUpdateComment: updateComment,
};

const CommentForm = connect(
  mapStateToProps,
  mapDispatchToProps,
)(withStyles(styles)(CommentFormComponent));

export default CommentForm;
