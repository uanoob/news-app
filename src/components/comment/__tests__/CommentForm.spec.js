import React from 'react';
import { shallow } from 'enzyme';
import { CommentFormComponent } from '../CommentForm';

describe('CommentForm', () => {
  const mockCreateComment = jest.fn();
  const mockUpdateComment = jest.fn();
  const mockHandleDialogClick = jest.fn();
  const mockClearTextField = jest.fn();
  const initialState = {
    comment: '',
    commentInputTouched: false,
    commentInputValid: false,
  };
  const props = {
    classes: {
      container: '',
      textField: '',
    },
    articleId: '',
    onCreateComment: mockCreateComment,
    onUpdateComment: mockUpdateComment,
    userId: 'PropTypes.string.isRequired',
    userName: 'PropTypes.string.isRequired',
    dialog: false,
    handleDialogClick: mockHandleDialogClick,
    clearTextField: mockClearTextField,
    title: 'PropTypes.string.isRequired',
    description: 'PropTypes.string.isRequired',
    commentId: '',
    commentText: 'PropTypes.string',
  };
  describe('CommentFormComponent initial', () => {
    const wrapper = shallow(<CommentFormComponent {...props} />);
    it('render correctly CommentFormComponent', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('initialize CommentFormComponent with initial state', () => {
      expect(wrapper.state()).toEqual(initialState);
    });
  });
  describe('CommentFormComponent checked input', () => {
    const wrapper = shallow(<CommentFormComponent {...props} />);
    describe('valid comment text field input', () => {
      const comment = 'GraphQl for beginners';
      const event = {
        target: {
          value: comment,
        },
        preventDefault: () => {},
      };
      beforeEach(() => {
        wrapper.find('#comment-textfield-input').simulate('change', event);
      });
      it('updates comment field in state', () => {
        expect(wrapper.state('comment')).toEqual(comment);
      });
      it('updates commentInputTouched field in state', () => {
        expect(wrapper.state('commentInputTouched')).toEqual(true);
      });
      it('updates commentInputValid field in state', () => {
        expect(wrapper.state('commentInputValid')).toEqual(true);
      });
    });
    describe('invalid comment text field input', () => {
      const comment = 'g';
      const event = {
        target: {
          value: comment,
        },
        preventDefault: () => {},
      };
      beforeEach(() => {
        wrapper.find('#comment-textfield-input').simulate('change', event);
      });
      it('updates comment field in state', () => {
        expect(wrapper.state('comment')).toEqual(comment);
      });
      it('updates commentInputTouched field in state', () => {
        expect(wrapper.state('commentInputTouched')).toEqual(true);
      });
      it('updates commentInputValid field in state', () => {
        expect(wrapper.state('commentInputValid')).toEqual(false);
      });
    });
  });
  describe('CommentFormComponent submit', () => {
    const wrapper = shallow(<CommentFormComponent {...props} />);
    describe('when clicking the Send button and no commentId', () => {
      const event = { preventDefault: () => {} };
      wrapper.find('#comment-submit-button').simulate('click', event);
      it('calls the onCreateComment', () => {
        expect(mockCreateComment).toBeCalledTimes(1);
      });
    });
  });
  describe('CommentFormComponent submit + commentId', () => {
    const nextProps = {
      ...props,
      commentId: '123',
    };
    const wrapper = shallow(<CommentFormComponent {...nextProps} />);
    describe('when clicking the Send button and commentId', () => {
      const event = { preventDefault: () => {} };
      wrapper.find('#comment-submit-button').simulate('click', event);
      it('calls onUpdateComment', () => {
        expect(mockUpdateComment).toBeCalledTimes(1);
      });
    });
  });
});
