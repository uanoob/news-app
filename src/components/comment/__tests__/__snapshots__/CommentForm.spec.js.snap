// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm CommentFormComponent initial render correctly CommentFormComponent 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentFormComponent
    articleId=""
    classes={
      Object {
        "container": "",
        "textField": "",
      }
    }
    clearTextField={[MockFunction]}
    commentId=""
    commentText="PropTypes.string"
    description="PropTypes.string.isRequired"
    dialog={false}
    handleDialogClick={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onCreateComment={
      [MockFunction] {
        "calls": Array [
          Array [
            "",
            "",
            "PropTypes.string.isRequired",
            "PropTypes.string.isRequired",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onUpdateComment={
      [MockFunction] {
        "calls": Array [
          Array [
            "",
            "",
            "123",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    title="PropTypes.string.isRequired"
    userId="PropTypes.string.isRequired"
    userName="PropTypes.string.isRequired"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-labelledby": "form-dialog-title",
      "children": Array [
        <WithStyles(DialogTitle)
          id="form-dialog-title"
        >
          PropTypes.string.isRequired
        </WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)>
            PropTypes.string.isRequired
          </WithStyles(DialogContentText)>
          <TextField
            autoFocus={true}
            className=""
            error={false}
            fullWidth={true}
            id="comment-textfield-input"
            label="Your comment here"
            margin="normal"
            multiline={true}
            onChange={[Function]}
            required={false}
            rows="4"
            select={false}
            value="PropTypes.string"
            variant="outlined"
          />
        </WithStyles(DialogContent)>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="primary"
            onClick={[Function]}
          >
            Cancel
          </WithStyles(Button)>
          <WithStyles(Button)
            color="primary"
            disabled={true}
            id="comment-submit-button"
            onClick={[Function]}
            type="submit"
            variant="contained"
          >
            Send
            <pure(SendIcon)>
              send
            </pure(SendIcon)>
          </WithStyles(Button)>
        </WithStyles(DialogActions)>,
      ],
      "onClose": [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "open": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "PropTypes.string.isRequired",
          "id": "form-dialog-title",
        },
        "ref": null,
        "rendered": "PropTypes.string.isRequired",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(DialogContentText)>
              PropTypes.string.isRequired
            </WithStyles(DialogContentText)>,
            <TextField
              autoFocus={true}
              className=""
              error={false}
              fullWidth={true}
              id="comment-textfield-input"
              label="Your comment here"
              margin="normal"
              multiline={true}
              onChange={[Function]}
              required={false}
              rows="4"
              select={false}
              value="PropTypes.string"
              variant="outlined"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "PropTypes.string.isRequired",
            },
            "ref": null,
            "rendered": "PropTypes.string.isRequired",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoFocus": true,
              "className": "",
              "error": false,
              "fullWidth": true,
              "id": "comment-textfield-input",
              "label": "Your comment here",
              "margin": "normal",
              "multiline": true,
              "onChange": [Function],
              "required": false,
              "rows": "4",
              "select": false,
              "value": "PropTypes.string",
              "variant": "outlined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              Cancel
            </WithStyles(Button)>,
            <WithStyles(Button)
              color="primary"
              disabled={true}
              id="comment-submit-button"
              onClick={[Function]}
              type="submit"
              variant="contained"
            >
              Send
              <pure(SendIcon)>
                send
              </pure(SendIcon)>
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cancel",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Send",
                <pure(SendIcon)>
                  send
                </pure(SendIcon)>,
              ],
              "className": undefined,
              "color": "primary",
              "disabled": true,
              "id": "comment-submit-button",
              "onClick": [Function],
              "type": "submit",
              "variant": "contained",
            },
            "ref": null,
            "rendered": Array [
              "Send",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "send",
                  "className": undefined,
                },
                "ref": null,
                "rendered": "send",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-labelledby": "form-dialog-title",
        "children": Array [
          <WithStyles(DialogTitle)
            id="form-dialog-title"
          >
            PropTypes.string.isRequired
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)>
              PropTypes.string.isRequired
            </WithStyles(DialogContentText)>
            <TextField
              autoFocus={true}
              className=""
              error={false}
              fullWidth={true}
              id="comment-textfield-input"
              label="Your comment here"
              margin="normal"
              multiline={true}
              onChange={[Function]}
              required={false}
              rows="4"
              select={false}
              value="PropTypes.string"
              variant="outlined"
            />
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="primary"
              onClick={[Function]}
            >
              Cancel
            </WithStyles(Button)>
            <WithStyles(Button)
              color="primary"
              disabled={true}
              id="comment-submit-button"
              onClick={[Function]}
              type="submit"
              variant="contained"
            >
              Send
              <pure(SendIcon)>
                send
              </pure(SendIcon)>
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "onClose": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "open": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "PropTypes.string.isRequired",
            "id": "form-dialog-title",
          },
          "ref": null,
          "rendered": "PropTypes.string.isRequired",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(DialogContentText)>
                PropTypes.string.isRequired
              </WithStyles(DialogContentText)>,
              <TextField
                autoFocus={true}
                className=""
                error={false}
                fullWidth={true}
                id="comment-textfield-input"
                label="Your comment here"
                margin="normal"
                multiline={true}
                onChange={[Function]}
                required={false}
                rows="4"
                select={false}
                value="PropTypes.string"
                variant="outlined"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "PropTypes.string.isRequired",
              },
              "ref": null,
              "rendered": "PropTypes.string.isRequired",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "className": "",
                "error": false,
                "fullWidth": true,
                "id": "comment-textfield-input",
                "label": "Your comment here",
                "margin": "normal",
                "multiline": true,
                "onChange": [Function],
                "required": false,
                "rows": "4",
                "select": false,
                "value": "PropTypes.string",
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                onClick={[Function]}
              >
                Cancel
              </WithStyles(Button)>,
              <WithStyles(Button)
                color="primary"
                disabled={true}
                id="comment-submit-button"
                onClick={[Function]}
                type="submit"
                variant="contained"
              >
                Send
                <pure(SendIcon)>
                  send
                </pure(SendIcon)>
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cancel",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Send",
                  <pure(SendIcon)>
                    send
                  </pure(SendIcon)>,
                ],
                "className": undefined,
                "color": "primary",
                "disabled": true,
                "id": "comment-submit-button",
                "onClick": [Function],
                "type": "submit",
                "variant": "contained",
              },
              "ref": null,
              "rendered": Array [
                "Send",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "send",
                    "className": undefined,
                  },
                  "ref": null,
                  "rendered": "send",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
